R/utilities.r

```diff
@@ -299,6 +299,16 @@ find_args <- function(...) {
 # global data
 dummy_data <- function() data.frame(x = NA)
 
+with_seed <- function(seed, code) {
+  if (!is.null(seed)) {
+    old_seed <- get0(".Random.seed", globalenv())
```

Comment:
@yihui: knitr seems to cache/restore .Random.seed, too. The caching in knitr is explicit, but it looks like the restore happens implicitly. Could you please comment if my way of restoring the random seed is safe?

```r
with_seed <- function(seed, code) {
  if (!is.null(seed)) {
    old_seed <- get0(".Random.seed", globalenv())
    if (!is.null(old_seed)) {
      on.exit(assign(".Random.seed", old_seed, globalenv()), add = TRUE)
    }
  }
  code
}
```